Entities/Attachable.cpp

```diff
@@ -1,679 +1,450 @@
+
+	void Attachable::Update() {
+		if (!m_Parent) {
+			m_JointPos = m_Pos + RotateOffset(m_JointOffset);
+		} else {
+			m_PrevPos = m_Pos;
+			m_PrevVel = m_Vel;
+
+			m_JointPos = m_Parent->GetPos() + m_Parent->RotateOffset(GetParentOffset());
+			m_Pos = m_JointPos - RotateOffset(m_JointOffset);
+			m_Vel = m_Parent->GetVel();
+			m_Team = m_Parent->GetTeam();
+			if (InheritsHFlipped() != 0) { m_HFlipped = m_InheritsHFlipped == 1 ? m_Parent->IsHFlipped() : !m_Parent->IsHFlipped(); }
+			if (InheritsRotAngle()) { SetRotAngle(m_Parent->GetRotAngle() + m_InheritedRotAngleOffset); }
+
+			MOSRotating *rootParentAsMOSR = dynamic_cast<MOSRotating *>(GetRootParent());
+			if (rootParentAsMOSR && m_CollidesWithTerrainWhileAttached) {
+				// Note: This safety check exists to ensure the parent's AtomGroup contains this Attachable's Atoms in a subgroup. Hardcoded Attachables need this in order to work, since they're cloned before their parent's AtomGroup exists.
+				if (!rootParentAsMOSR->GetAtomGroup()->ContainsSubGroup(m_AtomSubgroupID)) { AddOrRemoveAtomsFromRootParentAtomGroup(true); }
+
+				float facingAngle = (m_HFlipped ? c_PI : 0) + GetRotAngle() * static_cast<float>(GetFlipFactor());
+				float parentFacingAngle = (m_Parent->IsHFlipped() ? c_PI : 0) + m_Parent->GetRotAngle() * static_cast<float>(m_Parent->GetFlipFactor());
+
+				if (!InheritsRotAngle()) {
+					Matrix atomRotationForSubgroup(facingAngle - parentFacingAngle);
+					Vector atomOffsetForSubgroup;
+					CalculateAtomOffsetForSubgroup(atomOffsetForSubgroup);
+					// Note: This line looks weird because CalculateAtomOffsetForSubgroup already adds JointOffset to the value, so we have to subtract it and added its rotated value instead.
+					atomOffsetForSubgroup += GetJointOffset() - (GetJointOffset() * atomRotationForSubgroup);
+					rootParentAsMOSR->GetAtomGroup()->UpdateSubAtoms(GetAtomSubgroupID(), atomOffsetForSubgroup, atomRotationForSubgroup);
+				}
+			}
+			m_DeepCheck = false;
+		}
 
-    Vector forces;
-    // Add up all the forces
-    for (deque<pair<Vector, Vector> >::iterator fItr = m_Forces.begin();
-         fItr != m_Forces.end(); ++fItr)
-        forces += fItr->first;
+		MOSRotating::Update();
 
-    // Joint held up, so act accordingly
-// TODO: Maybe not do this here, we might need the forces for other stuff?")
-    // Clear out forces after we've bundled them up.
-    m_Forces.clear();
+		// If we're attached to something, MovableMan doesn't own us, and therefore isn't calling our UpdateScripts method (and neither is our parent), so we should here.
+		if (m_Parent != nullptr && GetRootParent()->HasEverBeenAddedToMovableMan()) { UpdateScripts(); }
+	}
 
-    jointForces += forces;
-    return true;
-}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+	void Attachable::SetMass(const float newMass) {
+		float previousMassForUpdatingParent = m_Parent ? GetMass() : 0.0F;
```

Comment:
I debated on this a bunch and this is what I settled on because it takes up one less line haha, and I'm perfectly partial to ternaries.
Correct me if I'm wrong, but to do it with an if check, it'd have to be:
```
void Attachable::SetMass(const float newMass) {
	float previousMassForUpdatingParent = 0.0F;
	if (m_Parent) { previousMassForUpdatingParent = GetMass(); }
	MovableObject::SetMass(newMass);
	if (m_Parent) { m_Parent->UpdateAttachableAndWoundMass(previousMassForUpdatingParent, GetMass()); }
}
```
I can't move it into the later if check, cause SetMass updates its mass value, and it'd be getting the updated value by that point. I could just not do the ternary and always GetMass, but this avoids the extra function call haha

Reply:
This was my mistake. Don't know why, but I thought it would work if you moved it into the existing if check.